src/test/java/org/medicmobile/webapp/mobile/OpenSettingsDialogFragmentTest.java

```diff
@@ -0,0 +1,193 @@
+package org.medicmobile.webapp.mobile;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.CALLS_REAL_METHODS;
+import static org.mockito.Mockito.RETURNS_SMART_NULLS;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.withSettings;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.view.MotionEvent;
+import android.view.View;
+import android.view.View.OnTouchListener;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.MockSettings;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(sdk=28)
+public class OpenSettingsDialogFragmentTest {
```

Comment:
Do you think we need any test cases around the `TIME_BETWEEN_TABS` logic?  Not sure I have ever done time manipulation in jUnit tests, so can't say how much of a pain that might be. (That being said, 500ms is not really that long for a `Thread.sleep` in the test case... so maybe we could just manipulate time the "normal" way :smile: )

Reply:
This is nice, worked just fine. I think I'll keep using the `java.time.Clock` instead of the System's one... 
The funny thing is that `Clock myClock = Clock.fixed(Instant.ofEpochMilli(1), ZoneOffset.UTC);` has to be outside the `try` block otherwise there's a NPE when getting the millis with `java.time.Clock.systemUTC().millis()`.

Reply:
I've added the test case covering the tap timeout :) Thanks!